merkletree/pad.go

```diff
@@ -114,8 +114,9 @@ func (pad *PAD) GetSTR(epoch uint64) *SignedTreeRoot {
 func (pad *PAD) TB(key string, value []byte) (*TemporaryBinding, error) {
 	//FIXME: compute private index twice
 	//it would be refactored after merging VRF integration branch
-	index, _ := pad.computePrivateIndex(key)
-	tb := pad.currentSTR.sig
+	cStr := pad.currentSTR
```

Comment:
> Currently, when you call pad.Update(policies) it generates the next STR as if the passed in policies were the ones being used. But, if the private key changes, all these temporary bindings being handed out no longer have the right index.

Makes sense. Will change that.

> I think PAD needs a policies field reflecting the current policies in place. currentSTR may 

OK, I have to think about that (after the talks/program finished). But on first sight it makes sense to add a policies field to `PAD`.  

> currentSTR may be badly named in that it is really just the most recently signed tree root, whereas pad.tree is what's current and will be reflected in the next STR. 

What do you think about `lastSTR` or `latestSTR`?

> Then, when you call Update(policies), it should generate the STR with the policies that were actually used (pad.policies), and then swap the pad.policies with the ones being passed in to be used going forward. It can then reshuffle the pad.tree if the private key changed.

> Hopefully that makes sense.

Sounds reasonable.